src/DssSpell.t.sol

```diff
@@ -878,5 +930,151 @@ contract DssSpellTest is DssSpellTestBase {
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 1_000_000 * WAD;
+
+        // Increas line of RWA015 to be able to draw some DAI
+        GodMode.setWard(address(vat), address(this), 1);
+        vat.file("RWA015-A", "line", 3_500_000 * RAD);
+
+        // setting address(this) as operator
+        vm.store(address(rwa015AUrn), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+        assertEq(rwa015AUrn.can(address(this)), 1);
+
+        // 0 DAI in Output Conduit
+        assertEq(dai.balanceOf(address(rwa015AOutputConduit)), 0, "RWA015-A: Dangling Dai in input conduit before draw()");
+
+        // Draw 1m to test output conduit
+        rwa015AUrn.draw(drawAmount);
+
+        // DAI in Output Conduit
+        assertEq(dai.balanceOf(address(rwa015AOutputConduit)), drawAmount, "RWA015-A: Dai drawn was not send to the recipient");
+
+        // wards
+        GodMode.setWard(address(rwa015AOutputConduit), address(this), 1);
+        // may
+        rwa015AOutputConduit.mate(address(this));
+        assertEq(rwa015AOutputConduit.may(address(this)), 1);
+        rwa015AOutputConduit.hope(address(this));
+        assertEq(rwa015AOutputConduit.can(address(this)), 1);
+
+        rwa015AOutputConduit.kiss(address(this));
+        assertEq(rwa015AOutputConduit.bud(address(this)), 1);
+        rwa015AOutputConduit.pick(address(this));
+
+        uint256 pushAmount = drawAmount;
+        rwa015AOutputConduit.push(pushAmount);
+        rwa015AOutputConduit.quit();
+
+        assertEq(dai.balanceOf(address(rwa015AOutputConduit)), 0, "RWA015-A: Output conduit still holds Dai after quit()");
+        assertEq(psmGem.balanceOf(address(this)), pushAmount / daiPsmGemDiffDecimals, "RWA015-A: Psm GEM not sent to destination after push()");
+        assertEq(dai.balanceOf(address(rwa015AOutputConduit)), drawAmount - pushAmount, "RWA015-A: Dai not sent to destination after push()");
+    }
+
+    function testFailRWA015_A_OUTPUT_CONDUIT_PUSH_ABOVE_BALANCE() public {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 2_500_000 * WAD;
+
+        // setting address(this) as operator
+        vm.store(address(rwa015AUrn), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+
+        // Draw line DAI
+        rwa015AUrn.draw(drawAmount);
+
+        // auth
+        GodMode.setWard(address(rwa015AOutputConduit), address(this), 1);
+
+        // pick address(this)
+        rwa015AOutputConduit.hope(address(this)); // allow this to call pick
+        rwa015AOutputConduit.kiss(address(this)); // allow this to be picked
+        rwa015AOutputConduit.pick(address(this));
+
+        // push above balance
+        uint256 pushAmount = drawAmount + 1 * WAD;
+        rwa015AOutputConduit.mate(address(this)); // allow this to call push
+        rwa015AOutputConduit.push(pushAmount);    // fail
+    }
```

Comment:
I would remove this test from the spell. This is pretty much a unit test on the RwaSwapOutputConduit and we are already covered in the `rwa-toolkit` repo.

```suggestion
```